### PR TITLE
Release notes v1.25.0

### PR DESCRIPTION
**Pull request by aneshlya**

### CODE REVIEW

File: docs/ReleaseNotes.txt [hunk 1]
```diff
@@ -1,3 +1,113 @@
...
+
+Deprecated targets:
+
+ - `avx512knl-x16`, `gen9-x8` and `gen9-x16` targets are deprecated and will be
+   removed in future releases.
+ - Adjusted device names for `xelpg-x8` and `xelpg-x16` to `mtl-u` and `mtl-h`.
+
+Standard library:
+
+ - Fixed integer overflow in `exp` for -Inf, NaN and big numbers.
```
Comment (dbabokin) [inline docs/ReleaseNotes.txt:51]: The wording is a bit confusing, as it's floating point `exp()`, while overflow is integer.

File: docs/ReleaseNotes.txt [hunk 2]
```diff
@@ -1,3 +1,113 @@
...
+ - Added a function for the floating-point remainder of division (`fmod`) for
+   double and float types.
```
Comment (dbabokin) [inline docs/ReleaseNotes.txt:53]: What about "floating-point types" or just removing "double and float types"? It's double, float and float16.

File: docs/ReleaseNotes.txt [hunk 3]
```diff
@@ -1,3 +1,113 @@
+=== v1.25.0 === (TBD October 2024)
```
Comment (dbabokin) [inline docs/ReleaseNotes.txt:1]: Should we mention the changes in standard library implementation and new `stdlib.isph`?

Comment (nurmukhametov) [inline docs/ReleaseNotes.txt:1]: Need to fix `TBD` to the actual date.

File: docs/ReleaseNotes.txt [hunk 4]
```diff
@@ -1,3 +1,113 @@
...
+ - Extended template support for short vectors and array declarations.  You can
+   now use type and non-type parameters to specify the type and dimensions of
+   these types.
+ - Added error messages for variable declarations with incompatible type
+   specifiers.
+ - Supported typedef with same name as struct tag.
+
+Code generation:
+
+ - Switched to the NewGVN pass for global value numbering for CPU targets. This
```
Comment (nurmukhametov) [inline docs/ReleaseNotes.txt:23]: We need to remove `NewGVN` entry because of #3056.

**Review by nurmukhametov — CHANGES_REQUESTED** [review]

File: docs/ReleaseNotes.txt (hunk 2 above)
Comment (aneshlya) [reply docs/ReleaseNotes.txt]: Fixed.

File: docs/ReleaseNotes.txt (hunk 3 above)
Comment (aneshlya) [reply docs/ReleaseNotes.txt]: Updated

File: docs/ReleaseNotes.txt (hunk 4 above)
Comment (aneshlya) [reply docs/ReleaseNotes.txt]: Updated

**Review by dbabokin — APPROVED** [review]
Looks good to me.

**Pull request merged**